src/kw/kw.cpp

```diff
@@ -156,35 +224,164 @@ string KanoWorld::get_refresh_header(string token)
+    }
+
+    std::time_t duration = atol(expiration_date.c_str());
+
+    if (!duration) {
+        return false;
+    }
+
+    double seconds = difftime(now, duration);
+
+    if (verbose) {
+        char now_str[26];
+        ctime_r(static_cast<const time_t*>(&now), now_str);
+
+        char duration_str[26];
+        ctime_r(static_cast<const time_t*>(&duration), duration_str);
+
+        std::cout << ">>> Am_I_Logged_In() requested - Time: " << now << " - "
+                  << now_str << std::endl;
+
+        std::cout << ">>> Token expires: " << duration << " - "
+                  << duration_str << std::endl;
+        std::cout << ">>> Difference in seconds: " << seconds << std::endl;
+        std::cout << ">>> Token valid? " << (seconds < 0 ? "Yes" : "No")
+                  << std::endl;
+    }
+
+    return seconds < 0;
 }
 
 
 /**
  *   Not implemented yet
  */
-bool KanoWorld::am_i_logged_in(void)
+std::string KanoWorld::whoami()
 {
-    return false;
+    return "";
 }
 
 
 /**
- *   Not implemented yet
+ * \brief Returns the token from the server response
+ *
+ * \returns A string with the complete JWT token
+ */
+std::string KanoWorld::parse_token(
+        const std::shared_ptr<JSON_Value> res) const {
+    if (res) {
+        return json_object_dotget_string(json_object(res.get()), "data.token");
+    }
+
+    return "";
+}
+
+
+/**
+ * \brief Returns the token expiration date from the server response
+ *
+ * \returns A string with the Unix timestamp representing the token expiration date
+ *
  */
-string KanoWorld::whoami(void)
+std::string KanoWorld::parse_expiration_date(const std::shared_ptr<JSON_Value> res) const {
+
+    if (res) {
+        // For some reason, the Unix time returned by the server is divide by
+        // 1000 so we convert it back into a Unix time here. See:
+        // https://github.com/KanoComputing/kes-world-api/blob/develop/src/controllers/accounts.js#L35
+        int conversion = 1000;
+        string translate = json_object_dotget_string(
+            json_object(res.get()), "data.duration");
+
+        unsigned long long converted_time = stol(translate);
+        converted_time *= conversion;
+        return std::to_string(converted_time);
+    }
+
+    return "";
+}
+
+
+std::string KanoWorld::get_token() const {
+    return this->token;
+}
+
+
+std::string KanoWorld::get_expiration_date() const {
+    return this->expiration_date;
+}
+
+
+/**
+ * \brief Loads the cache data for the latest Kano World transaction
+ *
+ * The cache data contains the token and expiration timestamp
+ *
+ * \returns true if the data was successfully loaded.
+ */
+bool KanoWorld::load_data()
 {
-    return string("");
+    std::shared_ptr<JSON_Value> user_data(
+        json_parse_file(data_filename.c_str()),
+        json_value_free);
+
+    if (!user_data) {
+        return false;
+    }
+
+    token = json_object_get_string(json_object(user_data.get()), "token");
+    expiration_date = json_object_get_string(
```

Comment:
Convention: Either we use `this->` for members or we use a `m_` prefix.
```suggestion
    this->expiration_date = json_object_get_string(
```